In [8]:
import xarray as xr

ds = xr.open_dataset("../data/raw/sst/sst_bob.nc")

ds

<xarray.Dataset> Size: 243kB
Dimensions:                  (time: 1, latitude: 150, longitude: 200)
Coordinates:
  * time                     (time) datetime64[ns] 8B 2020-01-01
  * latitude                 (latitude) float64 1kB 5.05 5.15 ... 19.85 19.95
  * longitude                (longitude) float64 2kB 80.05 80.15 ... 99.85 99.95
Data variables:
    sea_surface_temperature  (time, latitude, longitude) float64 240kB ...
Attributes: (12/65)
    Conventions:                     CF-1.7,ACDD-1.3,ISO 8601
    acknowledgment:                  Please acknowledge the use of these data...
    cdm_data_type:                   grid
    cmems_product_id:                SST_GLO_PHY_L3S_MY_010_039
    comment:                         WARNING:Some applications are unable to ...
    contact:                         jean.francois.piolle@ifremer.fr;emmanuel...
    ...                              ...
    time_coverage_end:               20240101T000000Z
    time_coverage_resolution:        P1D
    time_coverage_start:             20231231T000000Z
    title:                           Merged collation of sea surface temperat...
    uuid:                            
    copernicusmarine_version:        2.4.1

In [10]:
ds["sea_surface_temperature"].attrs

{'authority': 'CF-1.7',
 'long_name': 'sea surface temperature',
 'standard_name': 'sea_surface_foundation_temperature',
 'type': 'foundation',
 'units': 'kelvin',
 'valid_max': np.int16(4500),
 'valid_min': np.int16(-300)}

###SST Dataset Idea
so,the sst dataset has the shape of (1,150,200) in time,lat,long and the sea_surface_temperature as the variable for the dataset.The whole dataset is in netCDF format,so we will have to use xarrays.

Dimensions:these are the actuals values(or rows as we say in pandas) of the dataset, which are time, latitude, and longitude. The time dimension has 1 value, latitude has 150 values, and longitude has 200 values.

Shape: The shape of the dataset is (1, 150, 200), which corresponds to the dimensions of time, latitude, and longitude respectively. This means that for each time point (in this case, just one), there are 150 latitude points and 200 longitude points, resulting in a grid of sea surface temperature values.

coordinates: The coordinates of the dataset are the actual values of time, latitude, and longitude. The time coordinate will have a single value (e.g., a specific date), while the latitude and longitude coordinates will have 150 and 200 values respectively, representing the spatial grid of the sea surface temperature data.

data variables: The main data variable in the SST dataset is the sea_surface_temperature, which contains the temperature values at each grid point defined by the latitude and longitude coordinates. This variable will have a shape of (1, 150, 200), corresponding to the dimensions of time, latitude, and longitude.So,it is basically the output value that we are interested in analyzing or visualizing.

attributes: The attributes of the dataset provide additional metadata about the data, such as units, descriptions, and any other relevant information. For example, the sea_surface_temperature variable may have an attribute indicating that the temperature is measured in degrees Celsius. Other attributes may include information about the source of the data, the time range covered by the dataset, and any processing steps that were applied to the data.
we can also find attributes for a particular data variable, such as the sea_surface_temperature, which may include details about the measurement method, calibration, or any quality control procedures that were applied to ensure the accuracy of the data. These attributes are important for understanding the context and reliability of the dataset, and they can help guide further analysis or interpretation of the sea surface temperature values.

In [11]:
import xarray as xr

ds1 = xr.open_dataset("../data/raw/ssh/ssh_bob.nc")


ds1

<xarray.Dataset> Size: 39kB
Dimensions:    (time: 1, latitude: 60, longitude: 80)
Coordinates:
  * time       (time) datetime64[ns] 8B 2020-01-01
  * latitude   (latitude) float32 240B 5.125 5.375 5.625 ... 19.38 19.62 19.88
  * longitude  (longitude) float32 320B 80.12 80.38 80.62 ... 99.38 99.62 99.88
Data variables:
    sla        (time, latitude, longitude) float64 38kB ...
Attributes: (12/43)
    Conventions:                     CF-1.6
    Metadata_Conventions:            Unidata Dataset Discovery v1.0
    cdm_data_type:                   Grid
    comment:                         Sea Surface Height measured by Altimetry...
    contact:                         http://climate.copernicus.eu/c3s-user-se...
    creator_email:                   http://climate.copernicus.eu/c3s-user-se...
    ...                              ...
    time_coverage_duration:          P1D
    time_coverage_end:               2023-12-31T12:00:00Z
    time_coverage_resolution:        P1D
    time_coverage_start:             2023-12-30T12:00:00Z
    title:                           DT merged two satellites Global Ocean Gr...
    copernicusmarine_version:        2.4.1

In [12]:
import xarray as xr

ds2 = xr.open_dataset("../data/raw/glorys/glorys_bob.nc")

ds2

<xarray.Dataset> Size: 13MB
Dimensions:    (time: 1, depth: 36, latitude: 181, longitude: 241)
Coordinates:
  * time       (time) datetime64[ns] 8B 2020-01-01
  * depth      (depth) float32 144B 0.494 1.541 2.646 ... 763.3 902.3 1.062e+03
  * latitude   (latitude) float32 724B 5.0 5.083 5.167 5.25 ... 19.83 19.92 20.0
  * longitude  (longitude) float32 964B 80.0 80.08 80.17 ... 99.83 99.92 100.0
Data variables:
    thetao     (time, depth, latitude, longitude) float64 13MB ...
Attributes: (12/25)
    Conventions:               CF-1.4
    bulletin_date:             2021-07-07 00:00:00
    bulletin_type:             operational
    comment:                   CMEMS product
    domain_name:               GL12
    easting:                   longitude
    ...                        ...
    references:                http://www.mercator-ocean.fr
    source:                    MERCATOR GLORYS12V1
    title:                     daily mean fields from Global Ocean Physics An...
    z_max:                     5727.9169921875
    z_min:                     0.49402499198913574
    copernicusmarine_version:  2.4.1

In [13]:
import xarray as xr

ds3 = xr.open_dataset("../data/raw/winds/winds_bob.nc")

ds3

<xarray.Dataset> Size: 77kB
Dimensions:         (time: 1, latitude: 60, longitude: 80)
Coordinates:
  * time            (time) datetime64[ns] 8B 2020-01-01
  * latitude        (latitude) float32 240B 5.125 5.375 5.625 ... 19.62 19.88
  * longitude       (longitude) float32 320B 80.12 80.38 80.62 ... 99.62 99.88
Data variables:
    eastward_wind   (time, latitude, longitude) float64 38kB ...
    northward_wind  (time, latitude, longitude) float64 38kB ...
Attributes: (12/30)
    Conventions:                      CF-1.6
    comment:                          Orbit period and inclination are consta...
    contents:                         ovw
    creation_date:                    2025-02-26
    creation_time:                    05:02:55
    equator_crossing_date:            2025-02-24
    ...                               ...
    start_time:                       00:00:00
    stop_date:                        2025-02-25
    stop_time:                        23:59:56
    title:                            Global Ocean - Wind - METOP-B ASCAT - 2...
    title_short_name:                 ASCATB-L3-25km
    copernicusmarine_version:         2.4.1

In [14]:
import xarray as xr

ds4 = xr.open_dataset("../data/raw/currents/currents_bob.nc")

ds4

<xarray.Dataset> Size: 77kB
Dimensions:    (time: 1, depth: 1, latitude: 60, longitude: 80)
Coordinates:
  * time       (time) datetime64[ns] 8B 2020-01-01
  * depth      (depth) int16 2B 0
  * latitude   (latitude) float32 240B 5.125 5.375 5.625 ... 19.38 19.62 19.88
  * longitude  (longitude) float32 320B 80.12 80.38 80.62 ... 99.38 99.62 99.88
Data variables:
    uo         (time, depth, latitude, longitude) float64 38kB ...
    vo         (time, depth, latitude, longitude) float64 38kB ...
Attributes: (12/18)
    Conventions:               CF-1.6
    contact:                   servicedesk.cmems@mercator-ocean.eu
    creation_date:             2024-05-04 14:09:15
    geospatial_lat_units:      degrees_north
    geospatial_lon_units:      degrees_east
    geospatial_vertical_max:   0.0
    ...                        ...
    lon_min:                   -179.875
    lon_res:                   0.25
    product_version:           4.0
    references:                Mulet S., Rio M.-H., Etienne H., Artana C., Ca...
    title:                     Daily mean total surface and 15m velocities
    copernicusmarine_version:  2.4.1